src/query/helper.rs

```diff
@@ -532,3 +555,32 @@ pub(crate) fn unpack_table_alias(table_ref: &TableRef) -> Option<DynIden> {
         | TableRef::DatabaseSchemaTableAlias(_, _, _, alias) => Some(SeaRc::clone(alias)),
     }
 }
+
+pub(crate) fn cast_enum_as_text<C>(expr: Expr, col: &C) -> SimpleExpr
+where
+    C: ColumnTrait,
+{
+    cast_enum_text_inner(expr, col, |col, _| col.as_enum(Alias::new("text")))
+}
+
+pub(crate) fn cast_text_as_enum<C>(expr: Expr, col: &C) -> SimpleExpr
+where
+    C: ColumnTrait,
+{
+    cast_enum_text_inner(expr, col, |col, enum_name| {
+        col.as_enum(Alias::new(enum_name))
+    })
+}
+
+fn cast_enum_text_inner<C, F>(expr: Expr, col: &C, f: F) -> SimpleExpr
+where
+    C: ColumnTrait,
+    F: Fn(Expr, &String) -> SimpleExpr,
```

Comment:
Same here, I might have lost the context, but why are we passing enum name around as String? Can we pass around DynIden instead?

Reply:
The `String` comes from `ColumnType::Enum`
- https://github.com/SeaQL/sea-orm/blob/87f7891f175d662280ede2a234421bedac9507ca/src/entity/column.rs#L81

I rewrite it into `F: Fn(Expr, DynIden) -> SimpleExpr`, but still, we need to wrap the `String` in `DynIden` like `SeaRc::new(Alias::new(enum_name))`

Reply:
Um... I think we should have used an Iden here. Second question is, how often is this `ColumnType` object being cloned? If it is being cloned frequently then we'd better `Arc` it.